### PR TITLE
[RDY] Switch from a GraphStream backend to a Neo4j backend

### PR DESCRIPTION
**Pull request by gfokkema**
Remove all GraphStream related files and dependencies.

Define a Graph interface, with a Neo4j implementation.
Implement topological sorting using a depth first traversal.
Define getRank, which uses topological sort to calculate the longest paths for every node.

### CODE REVIEW

**Comment by Balletie** [comment]
I want people's thoughts on using an off-heap LongSet: this is a set for large amounts of data, and is not garbage collected (hence "off-heap").
The LongSet is closed by a try-with-resources statement, which frees the allocated memory.

**Comment by Unia** [comment]
While I must admit I know little of Neo4j, the code by itself looks good to me.

File: src/main/java/nl/tudelft/dnainator/graph/Neo4jGraphDatabase.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+		Result res = service.execute(GET_ROOT);
+		roots = res.columnAs("s");
+		return roots;
+	}
+
+	private Node getRoot() {
+		return rootIterator().next();
+	}
+
+	private static final int INIT_CAP = 4096;
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/graph/Neo4jGraphDatabase.java]: Not sure if this should be here or directly under the class declaration

**Comment by pvanagtmaal** [comment]
Apart from the very minor line comment, LGTM.

File: src/main/java/nl/tudelft/dnainator/ui/services/FileLoadService.java [hunk 2]
```diff
@@ -80,15 +77,15 @@ public final File getEdgeFile() {
 		return new Task<Graph>() {
```
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/services/FileLoadService.java]: Should be Task<Void>.

**Pull request merged**